client/my-sites/hosting/main.js

```diff
@@ -29,119 +30,150 @@ import { getAutomatedTransferStatus } from 'state/automated-transfer/selectors';
 import isAutomatedTransferActive from 'state/selectors/is-automated-transfer-active';
 import { transferStates } from 'state/automated-transfer/constants';
 import { requestSite } from 'state/sites/actions';
+import FeatureExample from 'components/feature-example';
 
 /**
  * Style dependencies
  */
 import './style.scss';
 
-const Hosting = ( {
-	canViewAtomicHosting,
-	clickActivate,
-	isDisabled,
-	isTransferring,
-	requestSiteById,
-	siteId,
-	siteSlug,
-	translate,
-	transferState,
-} ) => {
-	if ( ! canViewAtomicHosting ) {
-		return null;
+class Hosting extends Component {
+	state = {
+		clickOutside: false,
+	};
+
+	handleClickOutside( event ) {
+		const { COMPLETE } = transferStates;
+		const { isTransferring, transferState } = this.props;
+
+		if ( isTransferring && COMPLETE !== transferState ) {
+			event.preventDefault();
+			event.stopImmediatePropagation();
+			this.setState( { clickOutside: true } );
+		}
 	}
 
-	const getAtomicActivationNotice = () => {
-		const { COMPLETE, FAILURE } = transferStates;
+	render() {
+		const {
+			canViewAtomicHosting,
+			clickActivate,
+			isDisabled,
+			isTransferring,
+			requestSiteById,
+			siteId,
+			siteSlug,
+			translate,
+			transferState,
+		} = this.props;
 
-		// Transfer in progress
-		if (
-			( isTransferring && COMPLETE !== transferState ) ||
-			( isDisabled && COMPLETE === transferState )
-		) {
-			requestSiteById( siteId );
+		if ( ! canViewAtomicHosting ) {
+			return null;
+		}
 
-			return (
-				<>
-					<Notice
-						className="hosting__activating-notice"
-						status="is-info"
-						showDismiss={ false }
-						text={ translate( 'Please wait while we activate the hosting features.' ) }
-						icon="sync"
-					/>
-				</>
+		const getAtomicActivationNotice = () => {
```

Comment:
Optional: this could be now a method of the component, which should improve a little bit the readability. Fine to do this in a follow up PR.

Reply:
The reason I left them in there is to be able to easily move back to functional once we have a `useEffect` implementation of the `click outside` events.